src/Pool.ts

```diff
@@ -0,0 +1,86 @@
+import { catchError, combineLatest, map, of, switchMap, timeout } from 'rxjs'
+import type { Centrifuge } from './Centrifuge.js'
+import { Entity } from './Entity.js'
+import { PoolDomain } from './PoolDomain.js'
+
+export class Pool extends Entity {
+  constructor(_root: Centrifuge, public id: string) {
+    super(_root, ['pool', id])
+  }
+
+  domains() {
+    return this._query(null, () => {
+      return of(
+        this._root.chains.map((chainId) => {
+          return new PoolDomain(this._root, this, chainId)
+        })
+      )
+    })
+  }
+
+  // return this._query(['tranches'], () => {
+  //   return this._root
+  //     ._getSubqueryObservable<{ pool: { tranches: { nodes: { trancheId: string }[] } } }>(
+  //       `query($poolId: String!) {
+  //         pool(id: $poolId) {
+  //           tranches {
+  //             nodes {
+  //               trancheId
+  //             }
+  //           }
+  //         }
+  //       }`,
+  //       {
+  //         poolId: this.id,
+  //       }
+  //     )
+  //     .pipe(
+  //       map((data) => {
+  //         return data.pool.tranches.nodes.map((node) => node.trancheId)
+  //       })
+  //     )
+  // })
+  tranches() {
+    return this._root._querySubquery(
+      ['tranches', this.id],
+      `query($poolId: String!) {
+        pool(id: $poolId) {
+          tranches {
+            nodes {
+              trancheId
+            }
+          }
+        }
+      }`,
+      {
+        poolId: this.id,
+      },
+      (data: { pool: { tranches: { nodes: { trancheId: string }[] } } }) => {
+        return data.pool.tranches.nodes.map((node) => node.trancheId)
+      }
+    )
+  }
+
+  activeDomains() {
+    return this._query(null, () => {
+      return this.domains().pipe(
+        switchMap((domains) => {
+          return combineLatest(
+            domains.map((domain) =>
+              domain.isActive().pipe(
+                timeout(8000),
+                catchError(() => {
+                  return of(false)
+                })
+              )
+            )
+          ).pipe(
+            map((isActive) => {
+              return domains.filter((_, index) => isActive[index])
+            })
+          )
+        })
+      )
+    })
```

Comment:
would this fit better in the PoolDomain entity?

Reply:
still think this particular method doesn't belong here. Maybe we should rethink the PoolNetwork entity and figure out how it can handle all networks the pool is deployed on too. Possibly it's the name PoolNetwork that's just misleading in this case. Either way we should discuss how granular we want out entities to be and come up with a strategy together

Reply:
The PoolNetwork entity, as I imagine it, represents the presence of a pool on a single network. I see it being used to query if the pool is active, get the tranche token, get the liquidity pools for the pool on a specific network. As well as deploying the tranches/liquidity pools/etc. Everything that is cross-chain can live in the Pool itself. The word "network" maybe makes it sound a bit ambiguous, but it's a common term to mean a single chain. We can definitely brainstorm some ideas for how the entities may be laid out and how granular they should be.